test/e2e/garbage_collector.go

```diff
@@ -132,6 +132,15 @@ func verifyRemainingDeploymentsAndReplicaSets(
 		ret = false
 		By(fmt.Sprintf("expected %d Deploymentss, got %d Deployments", deploymentNum, len(deployments.Items)))
 	}
+	pods, err := clientSet.CoreV1().Pods(f.Namespace.Name).List(metav1.ListOptions{})
+	if err != nil {
+		return false, fmt.Errorf("Failed to list pods: %v", err)
+	}
+	if len(rsNum) == 0 && len(pods) != podNum {
```

Comment:
podNum is already passed as zero. It seems you only need `if len(pods.Items) != podNum {`